From 6832d721c16b7b9a6d55acd421c110409bc4a410 Mon Sep 17 00:00:00 2001
From: Andy Holmes <andrew.g.r.holmes@gmail.com>
Date: Tue, 31 Oct 2023 13:22:33 -0700
Subject: [PATCH] online-accounts: free window handle when unexporting

When unexporting the window handle be sure to free the string as well.
---
 panels/online-accounts/cc-online-accounts-panel.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/panels/online-accounts/cc-online-accounts-panel.c b/panels/online-accounts/cc-online-accounts-panel.c
index a88eb2836c..139f38e178 100644
--- a/panels/online-accounts/cc-online-accounts-panel.c
+++ b/panels/online-accounts/cc-online-accounts-panel.c
@@ -498,6 +498,8 @@ unexport_window_handle (CcOnlineAccountsPanel *self)
       gdk_wayland_toplevel_unexport_handle (GDK_TOPLEVEL (surface));
     }
 #endif
+
+  g_clear_pointer (&self->window_export_handle, g_free);
 }
 
 static void
-- 
GitLab

